client/executor/wasmtime/src/runtime.rs

```diff
@@ -119,20 +120,68 @@ impl WasmInstance for WasmtimeInstance {
 	}
 }
 
+/// Prepare a directory structure and a config file to enable wasmtime caching.
+///
+/// In case of an error the caching will not be enabled.
+fn setup_wasmtime_caching(
+	cache_path: &Path,
+	config: &mut Config,
+) -> std::result::Result<(), String> {
+	use std::fs;
+
+	let wasmtime_cache_root = cache_path.join("wasmtime");
+	fs::create_dir_all(&wasmtime_cache_root)
+		.map_err(|err| format!("cannot create the dirs to cache: {:?}", err))?;
+
+	// Canonicalize the path after creating the directories.
+	let wasmtime_cache_root = wasmtime_cache_root
+		.canonicalize()
+		.map_err(|err| format!("failed to canonicalize the path: {:?}", err))?;
+
+	// Write the cache config file
+	let cache_config_path = wasmtime_cache_root.join("cache-config.toml");
+	let config_content = format!(
+		"\
+[cache]
+enabled = true
+directory = \"{cache_dir}\"
+",
+		cache_dir = wasmtime_cache_root.display()
+	);
+	fs::write(&cache_config_path, config_content)
+		.map_err(|err| format!("cannot write the cache config: {:?}", err))?;
+
+	config
+		.cache_config_load(cache_config_path)
```

Comment:
I read in the docs that this needs a `config` feature enabled on wasmtime, but I can't see it anywhere. Am I missing something?

Reply:
(I assume you meant the `cache` feature), it is enabled by default as well.